components/dashboard/src/usage/download/DownloadUsage.tsx

```diff
@@ -0,0 +1,119 @@
+import { Dayjs } from "dayjs";
+import { useToast } from "../../components/toasts/Toasts";
+import { useCurrentOrg } from "../../data/organizations/orgs-query";
+import { SpinnerLoader } from "../../components/Loader";
+import { ReactComponent as DownloadIcon } from "../../icons/Download.svg";
+import { ReactComponent as ExclamationIcon } from "../../images/exclamation.svg";
+import { LinkButton } from "../../components/LinkButton";
+import { saveAs } from "file-saver";
+
+type Props = {
+    attributionId: AttributionId;
+    startDate: Dayjs;
+    endDate: Dayjs;
+};
+export const DownloadUsage: FC<Props> = ({ attributionId, startDate, endDate }) => {
+    const { data: org } = useCurrentOrg();
+    const { toast } = useToast();
+
+    const handleDownload = useCallback(async () => {
+        if (!org) {
+            return;
+        }
+
+        toast(
+            <DownloadUsageToast
+                orgName={org?.slug ?? org?.id}
+                attributionId={attributionId}
+                startDate={startDate}
+                endDate={endDate}
+            />,
+            {
+                autoHide: false,
+            },
+        );
+    }, [attributionId, endDate, org, startDate, toast]);
+
+    return (
+        <Button type="secondary" onClick={handleDownload} className="flex flex-row" icon={<DownloadIcon />}>
+            Export as CSV
+        </Button>
+    );
+};
+
+type DownloadUsageToastProps = Props & {
+    orgName: string;
+};
+
+const DownloadUsageToast: FC<DownloadUsageToastProps> = ({ attributionId, endDate, startDate, orgName }) => {
+    const queryArgs = useMemo(
+        () => ({
+            orgName,
+            attributionId: AttributionId.render(attributionId),
+            from: startDate.startOf("day").valueOf(),
+            to: endDate.endOf("day").valueOf(),
+        }),
+        [attributionId, endDate, orgName, startDate],
+    );
+    const { data, error, isLoading, abort, remove } = useDownloadUsageCSV(queryArgs);
+
+    const saveFile = useCallback(() => {
+        if (!data || !data.blob) {
+            return;
+        }
+
+        saveAs(data.blob, data.filename);
+    }, [data]);
+
+    useEffect(() => {
+        return () => {
+            abort();
+            remove();
+        };
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, []);
+
+    if (isLoading) {
+        return (
+            <div className="flex flex-row items-center space-x-2">
+                <SpinnerLoader small />
+                <span>Preparing usage export</span>
+            </div>
+        );
+    }
+
+    if (error) {
+        return (
+            <div>
+                <div className="flex flex-row items-start space-x-2">
+                    <ExclamationIcon className="mt-1 w-4 h-4" />
+                    <span>Error exporting your usage data:</span>
+                </div>
+                <pre className="mt-2 whitespace-normal text-sm">{error.message}</pre>
+            </div>
```

Comment:
**nitpick**: Saw the error toast on the screenshots and wanted to add some UX comments in case we could squeeze these changes here:
1. We could keep the toast simple and different from the alert component by dropping the icon here.
2. When designing a toasts and alerts, there's always the case where these two collide and become one component with two variants (inline, toast). But this is food for thought and out of the scope here.
3. If you'd prefer to keep the icon, could we align the text below the text so that the icon on the left stands out as we do with the alert component?
4. Could we also make sure the close button is aligned to the top, on the same horizontal line as the icon?

Reply:
Gotcha, if we can I'd like to keep the icon for now, as it would be difficult to transition the toast to a red background if it didn't start that way. I think I've got what you described handled, does this look right:

<img width="460" alt="image" src="https://github.com/gitpod-io/gitpod/assets/367275/7392cf9d-c4ed-4828-93e1-a4aa8b88b63a">

<img width="466" alt="image" src="https://github.com/gitpod-io/gitpod/assets/367275/9acaf0b8-9482-4065-8e77-3b0605058c26">

Reply:
Perfect—that's exactly what I had in mind, see designs in https://github.com/gitpod-io/gitpod/pull/17791#discussion_r1213534618.

Leaving the decision to keep the icon or not to you. 🏓

Reply:
![perfection](https://github.com/gitpod-io/gitpod/assets/120486/56e8d5ff-4a7f-4ba2-a8fc-4cd7db0d01cf)

Reply:
dropped it for the completed state, left it for the error.